capa/features/extractors/dnfile/helpers.py

```diff
@@ -169,11 +174,15 @@ def get_dotnet_managed_imports(pe: dnfile.dnPE) -> Iterator[DnType]:
             TypeName (index into String heap)
             TypeNamespace (index into String heap)
     """
-    for (rid, row) in enumerate(iter_dotnet_table(pe, "MemberRef")):
-        if not isinstance(row.Class.row, dnfile.mdtable.TypeRefRow):
+    for (rid, member_ref) in iter_dotnet_table(pe, dnfile.mdtable.MemberRef.number):
+        member_ref = cast(dnfile.mdtable.MemberRefRow, member_ref)
```

Comment:
ah, cool, TIL about `cast`.

in other parts of capa i've been using the pattern `assert isinstance(foo, Bar)` to narrow the types so mypy is happy. `cast` seems to do about the same thing, though it doesn't do any runtime checking.

at your discretion, you could use `isinstance(...)` here or keep as is. i dont feel strongly either way.

Reply:
I have opted for `assert`. The runtime type checking outweighs the small performance boost we get with `cast` essentially boiling down to a NOP.